torch/_export/constraints.py

```diff
@@ -1,9 +1,31 @@
-from typing import Optional
+from typing import Optional, Callable, Union
 
+import torch
+from torch.types import Number, _complex
+from torch import SymInt, SymFloat
 from torch._dynamo import allow_in_graph
-from torch.fx.experimental.symbolic_shapes import constrain_range
+from torch.fx.experimental.symbolic_shapes import constrain_range_non_symint
 from torch.utils._sympy.value_ranges import ValueRangeError
 
+# `Scalar` type used in native_functions.ymal will be translated to `Union[Number, _complex]`
+# could cause type error during since `SymInt` or `SymFloat` will be used.
+# Here manually specify the type explicitly.
+sym_constrain_range: Callable[
+    [Union[Number, _complex, SymInt, SymFloat], Optional[int], Optional[int]],
+    None,
+] = torch.sym_constrain_range  # type: ignore[assignment]
+
+
+def _constrain_range(symbol, min: Optional[int] = None, max: Optional[int] = None):
+    if not isinstance(symbol, SymInt):
+        # This is needed to check if an input value (as a real numeric value, or
+        # an value derived from numeric input) for exporting is within range or
+        # not.
+        constrain_range_non_symint(symbol, min=min, max=max)
+        return
```

Comment:
Just use `else` instead of plain `return`?

Reply:
Umm, actually I don't follow the logic here. Should you make `constrain_range_non_symint` return a symint that you can then call `sym_constraint_range` on?

Reply:
A satisfactory way to resolve this would be to write a test that actually exercises this path.

Reply:
Use the following logic as example:
```
def f(x):
    a = x.item()
    constrain_as_size(a, 4, 7)
    return torch.empty((a, 4))
```
The goal here is that if user passes in tensor like `torch.tensor([20])` during export, it will immediately fail since it's out of range. And based on my observation there is some round (I think it's the first round) of parsing, `20` (as `int`) will be passed in as `symbol` and the purpose of this logic is to test on this case.

> Should you make constrain_range_non_symint return a symint that you can then call sym_constraint_range on
Can try it out.

> A satisfactory way to resolve this would be to write a test that actually exercises this path.
This case is tested in the newly added test (`torch._export.export(f, (torch.tensor([20]),))`)